src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -85,15 +87,20 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             return result.Return;
         }
 
-        private async Task DelayUntilFunctionDispatcherInitialized()
+        private async Task DelayUntilFunctionDispatcherInitializedOrShutdown()
         {
             if (_functionDispatcher != null && _functionDispatcher.State == FunctionInvocationDispatcherState.Initializing)
             {
                 _logger.LogDebug($"functionDispatcher state: {_functionDispatcher.State}");
-                await Utility.DelayAsync(WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
+                await Utility.DelayAsync(_functionDispatcher.ErrorEventsThreshold * WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
                 {
                     return _functionDispatcher.State != FunctionInvocationDispatcherState.Initialized;
                 });
+
+                if (_functionDispatcher.State != FunctionInvocationDispatcherState.Initialized)
+                {
+                    _scriptJobHostEnvironment.Shutdown();
```

Comment:
Add error log indicating, _functionDispatcher initialization timedout and functions host is shutting down